### ISSUE TITLE
Use cloud config for storing logger configuration file

### ISSUE THREAD

**Issue by interair** [spring-cloud/spring-cloud-config#171]
Spring Cloud Config provides server and client-side support for externalized configuration in a distributed system. Important part of configuration is logger configuration. I suggest that storing logger configuration in cloud config fits to the main goal of project.

P.S. I think that will be good if cloud config will support any kind of configuration files in future including application aware.

**Comment by interair** [comment]
I tried to do a little improvements https://github.com/interair/spring-cloud-config/commit/0120b024ebb1da23997fbe3591b4c9ef0308e77e
In spring boot exists property:
logging.config= # location of config file (default classpath:logback.xml for logback)
it works only for classpath config files.
Was added new feature for loading logging config by specified path in cloud config server: 
it allows to update config file w/o redeploy service.

**Comment by dsyer** [comment]
I'm not really clear on what you patch is doing in detail. Can you clarify a bit?

Is this what you want to do: store an XML log configuration file (e.g. logback.xml) in the config server; download it in the client; copy it to the file system; re-initialize the logging system?

**Comment by interair** [comment]
Sure, maybe you will suggest much better solution:
Initially I wanted to add one more prefix implementation to [Resources](http://docs.spring.io/autorepo/docs/spring/4.2.x/spring-framework-reference/html/resources.html): example: **cloud**:data/config.xml and add to ConfigServicePropertySourceLocator and EnvironmentController ability to load any resources, but it is required a lot of changes in back-end.

I decided for the first tried to solve problem with storing/loading logging configuration.

According: [Appendix A. Common application properties](http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html) there is property:
logging.config= # location of config file
I decided to implement next behaviour:
1. On server: If such file exists in cloud config then try to load it (maybe will be good to add prefix cloud:logback.xml)
2. On client: reinitialize logger

```java
final ContextBase context = (ContextBase) StaticLoggerBinder.getSingleton().getLoggerFactory();
final JoranConfigurator configurator = new JoranConfigurator();
configurator.setContext(context);
context.reset();
configurator.doConfigure(new ByteArrayInputStream(loggerConfig.getBytes()));
```

I will be appreciated if you advice better solution
